.github/workflows/check-dev-containers.yml

```
@@ -0,0 +1,56 @@
+name: Check Dev Containers
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+  workflow_call:
+
+concurrency:
+  group: "${{ github.workflow }}-${{ github.head_ref || github.ref }}"
+  cancel-in-progress: true
+
+jobs:
+  changes:
+    name: Check if devcontainer has changed
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      devcontainer: ${{ steps.filter.outputs.devcontainer }}
+    steps:
+    - uses: actions/checkout@v4.1.1
+    - uses: dorny/paths-filter@v2.11.1
+      id: filter
+      with:
+        filters: |
+          devcontainer:
+            - '.devcontainer/**'
+            - .github/workflows/check-dev-containers.yml
+  container_base:
+    needs: changes
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4.1.1
+        if: ${{ needs.changes.outputs.devcontainer }}
```

I did some tests and it seems like specifying the conditions like this without a comparison lets the job run every time. Why not doing it the same way as in the documentation?
```suggestion
        if: ${{ needs.changes.outputs.devcontainer == 'true' || github.ref_name == 'main' || github.ref_type == 'tag'}}
```

---

I thought, GitHub's language would be similar "intelligent" than JavaScript ^^.

---

Me too 🤣